components/campaign/CampaignChaosBagView/index.tsx

```diff
@@ -0,0 +1,382 @@
+import React from 'react';
+import { keys, values } from 'lodash';
+import { BackHandler, Button, Platform, ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import { EventSubscription, Navigation } from 'react-native-navigation';
+
+import { t } from 'ttag';
+import { iconsMap } from '../../../app/NavIcons';
+import { NavigationProps } from '../../types';
+import { CHAOS_TOKENS, ChaosBag, ChaosTokenType } from '../../../constants';
+import { COLORS } from '../../../styles/colors';
+import { ChaosBagResults } from '../../../actions/types';
+import typography from '../../../styles/typography';
+import { EditChaosBagProps } from '../EditChaosBagDialog';
+import ChaosToken from '../ChaosToken';
+
+export interface CampaignChaosBagProps {
+    chaosBag: ChaosBag;
+    chaosBagResults: ChaosBagResults;
+    updateChaosBag: (chaosBag: ChaosBag) => void;
+    trackDeltas?: boolean;
+}
+
+interface State {
+    chaosBagResults: ChaosBagResults;
+    visible: boolean;
+    iconKey?: ChaosTokenType;
+}
+
+type Props = CampaignChaosBagProps & NavigationProps;
+
+// TODO: need to change it to pull chaosBag from redux state (pass the campaignId, and fetch campaign + get it from redux)
```

Comment:
Happy to help with this, or I can walk you through the process. The `CampaignDrawWeaknessDialog` is probably a good example of one that updates a piece of a campaign state in a similar way to this.  Our redux store for campaigns is just done by updating a 'partial' campaign, so you probably don't have to touch the literal redux store (just map the state out + use the existing `updateCampaign` action)